pageserver/src/page_service.rs

```diff
@@ -731,13 +734,14 @@ impl Connection {
 
     fn handle_pagerequests(&mut self, timelineid: ZTimelineId) -> Result<()> {
         // Check that the timeline exists
-        let pcache = page_cache::get_or_restore_pagecache(&self.conf, timelineid);
-        if pcache.is_err() {
+        let repository = page_cache::get_repository();
+        let timeline = repository.get_timeline(timelineid);
+        if timeline.is_err() {
             return Err(io::Error::new(
                 io::ErrorKind::InvalidInput,
                 format!("client requested pagestream on timeline {} which does not exist in page server", timelineid)));
         }
-        let pcache = pcache.unwrap();
+        let timeline = timeline.unwrap();
```

Comment:
`.is_err()` followed by `.unwrap()` can be improved. Maybe `map_err`, `unwrap_or` or `unwrap_or_else`?

```
let timeline = timeline.map_err(|_| {
    // build the error
}?;
```

Better yet, if `handle_pagerequests` returns `anyhow::Err` then all of that gets reduced to `.with_context(...)?`

Reply:
I'll use `map_err` for now. I agree that switching to `anyhow` would make sense here, but I'll leave that for a separate patch.